website/index.html

```diff
@@ -60,36 +38,62 @@ <h2>Echo</h2>
 <script type="text/javascript">if (!mobileConsole.status.initialized) {mobileConsole.init();}</script>
 -->
 
-<script type="text/javascript">console.log("Loading brython...")</script>
-<script src="https://cdnjs.cloudflare.com/ajax/libs/brython/3.9.1/brython.min.js"></script>
+<!-- It is important to remember that all text/python scripts are loaded and run in order AFTER all the javascript
+     scripts. You cannot interleave python/javascript scripts and expect them to run in order.
+-->
 
+<!-- Load the javascript loader functions -->
+<script src="loader.js"></script>
 <script type="text/javascript">
-  console.log("Loading brython_stdlib...");
-  document.getElementById("psychology").innerHTML = "Loading Standard Libraries...";
+  start_loader("Loading System...")
 </script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/brython/3.9.1/brython.min.js"></script>
+
+<!-- If you know your users will be on a slow system and will notice the download of brython and the std library
+     the first time then it may be worth showing an updated loader message. On subsequent runs these are likely to
+     be cached and will load instantly, leading to this message flashing too fast for the user to read.
+
+     Generally it may be better to leave a single "Loading System..." message and not try to update it at all.
+-->
+<!--<script type="text/javascript">-->
+<!--  update_loader_message("Loading Standard Library...")-->
+<!--</script>-->
```

Comment:
Different stages will take different amounts of time depending on caching.

The first time:

1. Load brython (may take a while on a slow connection?)
2. Load brython std. lib (make take a while on a slow connection)
3. Load the source of your main program (length depends on size of main)
4. Compile & run your main (will take a "long" time the first time as brython transpiles the code and stores the used std. library items to indexdb)

The second time:

1. Brython is cached, load is instant.
2. Std. Lib is cached, load is instant
3. Your program is cached, load is instant
4. Compile and run your main (faster but still the slowest step)

For me, messages like "Loading Brython" and "Loading Standard Library" don't mean much. I get the message that something is happening but that is all. If they are on the screen for 1/2 a second my feeling is just "was I supposed to read that?" On the second time around they might be on the screen for 1/10 of a second and its just a strobe effect so jarring instead of useful.

But we don't know if they are loading the first time or the second time (I guess we could check with a cookie...). In my testing it was less jarring to just have "Loading System..." on the screen with a rotating indicator and a pulse effect to show something was happening.

But I am sharing my experience, not arguing with you about it. I will make a new push with all messages activated.